### ISSUE TITLE
plugin_built_with_incompatible_erlang

### ISSUE THREAD

**Issue by softronsts** [gotthardp/rabbitmq-email#48]
Spent hours to compile at the end below is the result.

My env is 
Ubuntu 20.04
Erlang/OTP 24 [erts-12.2.1]
elixir 1.10.4
rabbitmq 3.9.13

Could you please advise the lib stack required to build rabbitmq_email?

Error:
Enabling plugins on node rabbit@rabbit:
rabbitmq_email
The following plugins have been configured:
  rabbitmq_email
  rabbitmq_management
  rabbitmq_management_agent
  rabbitmq_web_dispatch
Applying plugin configuration to rabbit@rabbit...
Stack trace:

** (CaseClauseError) no case clause matching: {:plugin_built_with_incompatible_erlang, 'rabbitmq_email'}
    (rabbitmqctl 3.8.0-dev) lib/rabbitmq/cli/plugins/plugins_helpers.ex:210: RabbitMQ.CLI.Plugins.Helpers.update_enabled_plugins/2
    (rabbitmqctl 3.8.0-dev) lib/rabbitmq/cli/plugins/plugins_helpers.ex:107: RabbitMQ.CLI.Plugins.Helpers.update_enabled_plugins/4
    (rabbitmqctl 3.8.0-dev) lib/rabbitmq/cli/plugins/commands/enable_command.ex:121: anonymous fn/6 in RabbitMQ.CLI.Plugins.Commands.EnableCommand.do_run/2
    (elixir 1.10.4) lib/stream.ex:1325: anonymous fn/2 in Stream.iterate/2
    (elixir 1.10.4) lib/stream.ex:1538: Stream.do_unfold/4
    (elixir 1.10.4) lib/stream.ex:1609: Enumerable.Stream.do_each/4
    (elixir 1.10.4) lib/stream.ex:956: Stream.do_enum_transform/7
    (elixir 1.10.4) lib/stream.ex:1609: Enumerable.Stream.do_each/4

{:case_clause, {:plugin_built_with_incompatible_erlang, 'rabbitmq_email'}}

**Comment by lukebakken** [comment]
Thanks for the information. I'd also like to know which commands you're running to build this plugin and then copy it your RabbitMQ 3.9.13 installation.

In general, you should build this plugin using the _minimum_ supported Erlang for the version of RabbitMQ you wish to use - https://www.rabbitmq.com/which-erlang.html

In this case, that means Erlang 23.2 and Elixir 1.10.4. I will make sure that the [README](https://github.com/gotthardp/rabbitmq-email/blob/master/README.md) reflects this. I also just realized I should improve the build instructions there.

I have attached a transcript of a successful build (truncated to remove excessive noise), installation, enablement and start of RabbitMQ using the `master` branch of this plugin. I'm running RabbitMQ using Erlang 24.2.2:

[transcript.txt](https://github.com/gotthardp/rabbitmq-email/files/8199418/transcript.txt)

Basically, these are the commands I ran in my clone of this repo:

```
git clean -xffd
asdf local erlang 23.2; asdf local elixir 1.10.4-otp-23
make
make dist
rsync -a plugins/rabbitmq_email-1.0.0 /home/bakkenl/rmq-server/rabbitmq_server-3.9.13/plugins/
rsync -a plugins/eiconv-1.0.0 /home/bakkenl/rmq-server/rabbitmq_server-3.9.13/plugins/
rsync -a plugins/gen_smtp-1.2.0-dev /home/bakkenl/rmq-server/rabbitmq_server-3.9.13/plugins/
```

**Comment by a-marcellini** [comment]
You are syncing the dev. version of gen_smtp, however in the readme you declare the dependency to the 1.1.1. 
What's more I'm not sure if your code acknowledges the mimemail encode modification: from list to map.
[here the code](https://github.com/gen-smtp/gen_smtp/blob/628863f0bfd4439678bf216da5c1f55626ec2a8e/src/mimemail.erl#L80)

I'm not an erlang expert, however, I think that the code doesn't use maps as parameter but it uses lists:

Look at rabbit_email_sender.erl row 22 as an example, 4th parameter: 
`Message = mimemail:encode({Type, Subtype, lists:foldr(fun set_header/2, [], Headers2), [], Payload}),`

**Comment by lukebakken** [comment]
@a-marcellini I noticed the `gen_smtp` dependency as well and was going to investigate it today. Thanks for pointing out the other line as well. In the future, please feel free to open pull requests if you find something to fix!

https://github.com/gen-smtp/gen_smtp/issues/205

**Comment by lukebakken** [comment]
@softronsts please see the build instructions here - https://github.com/gotthardp/rabbitmq-email#installation

**Comment by lukebakken** [comment]
https://github.com/gotthardp/rabbitmq-email/releases/tag/v1.0.1

**Comment by softronsts** [comment]
Thanks for your help.  Managed to install the plugin.
asdf erlang is a kill in ubuntu 20.04.  Pinned to specific version and got erlang working without asdf.
elixir should be handled via asdf.

**Comment by a-marcellini** [comment]
I'm unable to do a pull request.
I've just created a Issue (51) and a local branch (gh-51), which I cannot
push for lack of permissions:

***@***.*** rabbitmq-email (gh-51) $ git push --set-upstream
origin gh-51
Username for 'https://github.com': ***@***.***
Password for ***@***.***@github.com':
remote: Permission to gotthardp/rabbitmq-email.git denied to a-marcellini.
fatal: unable to access 'https://github.com/gotthardp/rabbitmq-email.git/':
The requested URL returned error: 403

Let me know if I can try again in the future
Regards
Alessandro

Il giorno mar 8 mar 2022 alle ore 14:53 Luke Bakken <
***@***.***> ha scritto:

**Comment by lukebakken** [comment]
@a-marcellini you'll find the GitHub docs helpful - https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests

To contribute to this repository, you _fork_ it to your GitHub account and then push your local branch to your fork. Then you can use the GitHub web site to open the pull request.

At any rate, I have fixed the issue and re-tagged version 1.0.1.

Thanks everyone!
